local.go

```diff
@@ -275,8 +275,17 @@ func checkConvergence(loc *Location, itertype IterationType, stats *Stats, setti
 	return NotTerminated
 }
 
-// evaluate evaluates the function and stores the answer in place
+// evaluate evaluates the function given by funcs at xNext, stores the answer
+// in location and updates stats. If the function does not support the
+// requested evalType, location and stats are not modified and ErrMismatch is
+// returned.
 func evaluate(funcs functions, funcInfo *FunctionInfo, evalType EvaluationType, xNext []float64, location *Location, stats *Stats) error {
+	if evalType == NoEvaluation {
```

Comment:
Yea, I was not sure about the right behavior in this case. I wanted NoEvaluation to be a no-op, but your suggestion is also quite (even more) reasonable: who wants a no-op can choose not call evaluate() when NoEvaluation. I will modify accordingly.